### PR TITLE
Update to new method names in gtsam SfmTrack and SfmData

### PR DESCRIPTION
**Pull request by akshay-krishnan**
This only changes some method names, for ex: number_measurements() - > numberMeasurements(), add_measurement() -> addMeasurement(), due to the change on gtsam. 

GTSAM PR: https://github.com/borglab/gtsam/pull/1082

### CODE REVIEW

**Comment by akshay-krishnan** [comment]
Need to update the GTSAM wheel in the CI, left a TODO in setup.py

**Comment by johnwlambert** [comment]
thanks for the PR, @akshay-krishnan. A few more names for us to update, according to the CI:

```
==================================== ERRORS ====================================
______________ ERROR collecting tests/test_two_view_estimator.py _______________
tests/test_two_view_estimator.py:[17](https://github.com/borglab/gtsfm/runs/5113140405?check_suite_focus=true#step:7:17): in <module>
    EXAMPLE_DATA = io_utils.read_bal(gtsam.findExampleDataFile(GTSAM_EXAMPLE_FILE))
gtsfm/utils/io.py:137: in read_bal
    num_images = sfm_data.number_cameras()
E   AttributeError: 'gtsam.gtsam.SfmData' object has no attribute 'number_cameras'
___________ ERROR collecting tests/bundle/test_bundle_adjustment.py ____________
tests/bundle/test_bundle_adjustment.py:15: in <module>
    EXAMPLE_DATA = io_utils.read_bal(gtsam.findExampleDataFile(GTSAM_EXAMPLE_FILE))
gtsfm/utils/io.py:137: in read_bal
    num_images = sfm_data.number_cameras()
E   AttributeError: 'gtsam.gtsam.SfmData' object has no attribute 'number_cameras'
_______________ ERROR collecting tests/common/test_gtsfm_data.py _______________
tests/common/test_gtsfm_data.py:[19](https://github.com/borglab/gtsfm/runs/5113140405?check_suite_focus=true#step:7:19): in <module>
    EXAMPLE_DATA = io_utils.read_bal(gtsam.findExampleDataFile(GTSAM_EXAMPLE_FILE))
gtsfm/utils/io.py:1[37](https://github.com/borglab/gtsfm/runs/5113140405?check_suite_focus=true#step:7:37): in read_bal
    num_images = sfm_data.number_cameras()
E   AttributeError: 'gtsam.gtsam.SfmData' object has no attribute 'number_cameras'
______________ ERROR collecting tests/utils/test_serialization.py ______________
tests/utils/test_serialization.py:15: in <module>
    EXAMPLE_DATA = io_utils.read_bal(gtsam.findExampleDataFile(GTSAM_EXAMPLE_FILE))
gtsfm/utils/io.py:137: in read_bal
    num_images = sfm_data.number_cameras()
E   AttributeError: 'gtsam.gtsam.SfmData' object has no attribute 'number_cameras'
=============================== warnings summary ===============================
```

File: .github/scripts/setup.sh [hunk 1]
```diff
@@ -8,10 +8,10 @@ echo "Running .github/scripts/setup.sh..."
 conda init
 conda info --envs
 
-wget -O 2021_09_02_gtsam_python38_wheel.zip --no-check-certificate "https://drive.google.com/uc?export=download&id=1GSf3NxabSCsu2nPcBUuEhhVq_m1FvdY5"
+wget https://github.com/johnwlambert/gtsfm-datasets-mirror/releases/download/v4.2a5/gtsam-4.2a5-cp38-cp38-manylinux2014_x86_64.whl.zip
 
-unzip 2021_09_02_gtsam_python38_wheel.zip
-pip install gtsam-4.1.0-cp38-cp38-manylinux2014_x86_64.whl
+unzip gtsam-4.2a5-cp38-cp38-manylinux2014_x86_64.whl.zip
+pip install gtsam-4.2a5-cp38-cp38-manylinux2014_x86_64.whl
```
Comment (ayushbaid) [inline .github/scripts/setup.sh:14]: nit: you can export the name to make it easier from next time.

**Review by ayushbaid — APPROVED** [review]
You might also need to update the wheel links in the README.

File: .github/scripts/setup.sh (hunk 1 above)
Comment (akshay-krishnan) [reply .github/scripts/setup.sh]: udpated, hopefully the change is what you meant.

**Review by johnwlambert — APPROVED** [review]
LGTM, very nice. Thanks Akshay

**Pull request merged**